node/impl/full/state.go

```diff
@@ -1558,11 +1558,18 @@ func (a *StateAPI) StateMinerInitialPledgeCollateral(ctx context.Context, maddr
 		return types.EmptyInt, xerrors.Errorf("getting circulating supply: %w", err)
 	}
 
+	epochsSinceRampStart, rampDurationEpochs, err := a.getPledgeRampParams(ctx, ts.Height(), state)
+	if err != nil {
+		return types.EmptyInt, xerrors.Errorf("getting pledge ramp params: %w", err)
+	}
+
 	initialPledge, err := rewardState.InitialPledgeForPower(
 		sectorWeight,
 		pledgeCollateral,
 		powerSmoothed,
 		circSupply.FilCirculating,
+		epochsSinceRampStart,
```

Comment:
I'll advise again that it would be simpler to just have a function `max(InitialPledgeForPower(..., -1,0,...), InitialPledgeForPower(..., 0,0,...)` that takes max before and after ramp that you can use for these Miner APIs.

But what you have looks fine too.

Reply:
And I guess this is better for testing